### PR TITLE
Update the image configuration

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request updates the image configuration to both modernize it and improve platform support.

## 💭 Motivation and context ##

Modernizing the Docker image configuration will help with any necessary maintenance and development in the future. It also aligns the image with Docker best practices such as strict version pinning and multi-stage builds. Using a multi-stage build also allows us to restore support for non-`amd64`/`arm64` platforms.

## 🧪 Testing ##

Automated tests pass. I am able to use the latest pre-release to generate an appropriate `code.json` (barring some issues with [llnl-scraper](https://pypi.org/project/llnl-scraper/) that will be resolved separately).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Approved, but I would like to see these improvements (including the `Pipfile` foo) make their way into the skeleton before we apply them manually to too many more repos.

File: Dockerfile [hunk 1]
```diff
@@ -9,53 +17,87 @@ FROM python:3.7.11-slim-stretch
...
+# python3 in the venv to the Python binary in this image
+COPY --from=compile-stage --chown=${CISA_USER}:${CISA_GROUP} ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+RUN ln -sf "$(command -v python3)" "${VIRTUAL_ENV}"/bin/python3
+ENV PATH="${VIRTUAL_ENV}/bin:$PATH"
+
+
+# Copy in the necessary files
+COPY --chown=${CISA_USER}:${CISA_GROUP} src/update.sh src/email-update.py src/body.txt src/body.html ${CISA_HOME}/
+
+# Prepare to Run
```
Comment (dav3r) [inline Dockerfile]: Remove unnecessary capitalization:
```suggestion
# Prepare to run
```

File: src/config.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[net]
+# Configure Cargo to use the git CLI instead of the libgit2 library. There is
+# an issue with 32-bit (non-x86) platforms when ligbit2 tries to pull down the
```
Comment (dav3r) [inline src/config.toml]: Fix typo:
```suggestion
# an issue with 32-bit (non-x86) platforms when libgit2 tries to pull down the
```

**Review by dav3r — APPROVED** [review]
👍 👍

File: Dockerfile (hunk 1 above)
Comment (mcdonnnj) [reply Dockerfile]: The curse of copying things from other projects sometimes~

**Comment by mcdonnnj** [comment]
Disabled the snyk requirement because we've hit a test cap.

**Pull request merged**